datafusion/optimizer/src/utils.rs

```diff
@@ -424,31 +468,74 @@ mod tests {
     use super::*;
     use arrow::datatypes::DataType;
     use datafusion_common::Column;
-    use datafusion_expr::{binary_expr, col, lit, utils::expr_to_columns};
+    use datafusion_expr::{col, lit, utils::expr_to_columns};
     use std::collections::HashSet;
     use std::ops::Add;
 
     #[test]
-    fn combine_zero_filters() {
-        let result = combine_filters(&[]);
-        assert_eq!(result, None);
+    fn test_split_conjunction() {
```

Comment:
IMO, I think we can add test for `conjunction()`. At the same time, we can check the tree structure of this expression by using `match`.

```rs
expr is (A B) C;

/// using `match` to check.
match expr {
   And(
      And(
         B,
         C
      ),
      C
   )
}
```

[A, B, C ,D , E] -> (((A B) C) (D E)) is different from ((((A B) C) D) E).

we can see the result of `conjunction()` in the UT

Reply:
This was a great idea @jackwener  -- thank you. I added this test  09f3cf4e4

As part of writing tests, I also found that the API for `disjunction` was slightly different than `conjunction` so I made them the same as well.